tests/instances.tcl

```diff
@@ -93,10 +91,30 @@ proc spawn_instance {type base_port count {conf {}}} {
         close $cfg
 
         # Finally exec it and remember the pid for later cleanup.
-        set pid [exec_instance $type $cfgfile]
-        lappend ::pids $pid
+        set retry 100
+        while {$retry} {
+            set pid [exec_instance $type $cfgfile]
+
+            # Check availability
+            if {[server_is_up 127.0.0.1 $port 100] == 0} {
+                puts "Starting $type #$j at port $port failed, try another"
+                incr retry -1
+                set port [find_available_port $base_port $::redis_port_count]
+                set cfg [open $cfgfile a+]
+                if {$::tls} {
+                    puts $cfg "tls-port $port"
```

Comment:
```suggestion
                    puts $cfg "tls-port $port"
                    puts $cfg "port 0"
```

Reply:
Needed otherwise the instance will also attempt to listen on 6379 (and may fail).

Reply:
@yossigo this code appends to the exiting config file.
Redis will use the last one. 
The line you requested it already there. 
This is not an ideal solution, but I'm not expecting this retry loop to run many times (if at all)